### ISSUE TITLE
Feature Request: Auto Scaling Plans (Dynamic/Predictive Auto Scaling Groups)

### ISSUE THREAD

**Issue by kinghuang** [terraform-providers/terraform-provider-aws#6562]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Amazon recently introduced a predictive scaling capability in auto scaling groups. Auto Scaling Plans use dynamic scaling and predictive scaling to automatically scale ASGs. I believe a new resource is required to support this.

### New Resource(s)

* `aws_autoscaling_plan`

### Potential Terraform Configuration

```hcl
resource "aws_autoscaling_plan" "this" {
  name                    = "this-autoscaling-plan"
  min_capacity            = 1
  max_capacity            = 7
  predictive_scaling_mode = "forecast-and-scale"
  scalable_dimension      = "autoscaling:autoScalingGroup:DesiredCapacity"
  autoscaling_group_name  = "${aws_autoscaling_group.that.name}"
}
```

### References

* [AWS News Blog: New – Predictive Scaling for EC2, Powered by Machine Learning](https://aws.amazon.com/blogs/aws/new-predictive-scaling-for-ec2-powered-by-machine-learning/)
* [API Reference: CreateScalingPlan](https://docs.aws.amazon.com/autoscaling/plans/APIReference/API_CreateScalingPlan.html)
* [What Is AWS Auto Scaling? (Updated)](https://docs.aws.amazon.com/autoscaling/plans/userguide/what-is-aws-auto-scaling.html)

**Comment by darmawanaris** [comment]
@kinghuang have this feature already available on terraform ?

**Comment by kyprifog** [comment]
@kinghuang should this be closed?

**Comment by anGie44** [comment]
This feature has been merged and will release with `v3.12` of the Terraform AWS Provider, likely out this Thursday.
